### PR TITLE
[SYCL] Refactor include/sycl/properties/queue_properties.hpp

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Make sure the required amount of boilerplate code is minimal. Fixes
an issue with missing explicit instantiations - see new symbols in the
ABI test(s).

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
CUDA's

```
SYCL :: BFloat16/bfloat16_builtins.cpp
```

is known - https://github.com/intel/llvm/issues/8040.

**Comment by aelovikov-intel** [comment]
@KseniyaTikhomirova , ping.

File: sycl/test/abi/sycl_symbols_linux.dump [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # DO NOT EDIT IT MANUALLY. Refer to sycl/doc/developer/ABIPolicyGuide.md for more info.
 ################################################################################
 
-# RUN: env LLVM_BIN_PATH=%llvm_build_bin_dir %python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %sycl_libs_dir/libsycl.so
```
Comment (KseniyaTikhomirova) [inline sycl/test/abi/sycl_symbols_linux.dump:6]: why you removed it? I guess it was done like this for systems where python3 installed.

File: sycl/test/abi/sycl_symbols_windows.dump [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # DO NOT EDIT IT MANUALLY. Refer to sycl/doc/developer/ABIPolicyGuide.md for more info.
 ################################################################################
 
-# RUN: env LLVM_BIN_PATH=%llvm_build_bin_dir %python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %llvm_build_bin_dir/sycl6.dll
+# RUN: env LLVM_BIN_PATH=%llvm_build_bin_dir python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %llvm_build_bin_dir/sycl6.dll
```
Comment (KseniyaTikhomirova) [inline sycl/test/abi/sycl_symbols_windows.dump:6]: same

**Review by KseniyaTikhomirova — COMMENTED** [review]
To be honest I do not like that we introduce new macro. since we have many problems with user macro redefinitions I expected that we would prefer to avoid them. 
Although if we still was to add them - should we follow this guide https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#general-guidelines-1? It says:
"All identifiers used in llvm/sycl headers files must contain at least one lowercase letter due to avoid conflicts with user-defined macros".

File: sycl/test/abi/sycl_symbols_linux.dump (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/test/abi/sycl_symbols_linux.dump:6]: That's what the script did.

**Comment by aelovikov-intel** [comment]
> To be honest I do not like that we introduce new macro. since we have many problems with user macro redefinitions I expected that we would prefer to avoid them. Although if we still was to add them - should we follow this guide https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#general-guidelines-1? It says: "All identifiers used in llvm/sycl headers files must contain at least one lowercase letter due to avoid conflicts with user-defined macros".

Modified new macros' names. Not changing their arguments because they don't seem to be affected (and we don't have the rule universally applied to macros arguments in our codebase either) - https://godbolt.org/z/hsn1o6dT8 .

**Comment by KseniyaTikhomirova** [comment]
> > To be honest I do not like that we introduce new macro. since we have many problems with user macro redefinitions I expected that we would prefer to avoid them. Although if we still was to add them - should we follow this guide https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#general-guidelines-1? It says: "All identifiers used in llvm/sycl headers files must contain at least one lowercase letter due to avoid conflicts with user-defined macros".
> 
> Modified new macros' names. Not changing their arguments because they don't seem to be affected (and we don't have the rule universally applied to macros arguments in our codebase either) - https://godbolt.org/z/hsn1o6dT8 .

Could you please educate me where is the guideline about underscore usage?

**Review by KseniyaTikhomirova — APPROVED** [review]

**Comment by aelovikov-intel** [comment]
Failure `SYCL :: dword_atomic_smoke.cpp` is known - https://github.com/intel/llvm/issues/8098.

**Comment by aelovikov-intel** [comment]
@intel/llvm-gatekeepers , PR is ready.

**Pull request merged**